### ISSUE TITLE
ptp status script hook creates zombie processes

### ISSUE THREAD

**Issue by AleksanderZdunek** [bondagit/aes67-linux-daemon#85]
Child processes created in SessionManager::on_ptp_status_changed() are never reaped, leaving every execution of the ptp status script in a zombie state when it's finished.
See my output from ps
`ps -ef --forest`
```
root     2566380       1  0 12:38 ?        00:00:00 ./aes67-daemon -c daemon.conf
root     2566383 2566380  0 12:38 ?        00:00:00  \_ [ptp_status.sh] <defunct>
root     2566560 2566380  0 12:38 ?        00:00:00  \_ [ptp_status.sh] <defunct>
root     2570175 2566380  0 12:41 ?        00:00:00  \_ [ptp_status.sh] <defunct>
root     2570351 2566380  0 12:42 ?        00:00:00  \_ [ptp_status.sh] <defunct>
root     2571591 2566380  0 12:43 ?        00:00:00  \_ [ptp_status.sh] <defunct>
root     2571770 2566380  0 12:43 ?        00:00:00  \_ [ptp_status.sh] <defunct>
root     2571771       1  0 12:43 ?        00:00:00 speaker-test -D plughw:RAVENNA -r 48000 -c 2 -t sine
```

As a quick and dirty fix I added `waitpid(-1, NULL, WNOHANG);` to the worker loop in SessionManager::worker(). There are probably better ways of fixing this. One option that came to mind is double forking the child process and wait() for the outer child process in on_ptp_status_changed(). Another option that was suggested to me is to let boost::process deal with the issue.

**Comment by bondagit** [comment]
thanks for reporting this issue. Just apply the following patch:

```
diff --git a/daemon/main.cpp b/daemon/main.cpp
index 58fa162..885f840 100644
--- a/daemon/main.cpp
+++ b/daemon/main.cpp
@@ -91,6 +91,7 @@ int main(int argc, char* argv[]) {
   }
 
   signal(SIGINT, termination_handler);
+  signal(SIGCHLD, SIG_IGN);
   signal(SIGTERM, termination_handler);
 
   std::srand(std::time(nullptr));

```

**Comment by bondagit** [comment]
Fixed by daemon v1.5.1
